File: app/src/main/java/com/sjani/usnationalparkguide/Data/TrailDao.java
package com.sjani.usnationalparkguide.Data;


import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface TrailDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void save(TrailEntity trailEntity);

    @Query("SELECT * FROM trails")
    LiveData<List<TrailEntity>> getAllTrails();

    @Query("SELECT * FROM trails WHERE trail_id = :trailId")
    LiveData<TrailEntity> getTrail(String trailId);

    @Query("DELETE FROM trails")
    void clearTable();
}

File: app/src/main/java/com/sjani/usnationalparkguide/data/dao/TrailDao.kt
package com.sjani.usnationalparkguide.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.sjani.usnationalparkguide.data.entity.TrailEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface TrailDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun save(trailEntity: TrailEntity)
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun saveAll(trails: List<TrailEntity>)
    
    @Query("SELECT * FROM trails")
    fun getAllTrails(): Flow<List<TrailEntity>>
    
    @Query("SELECT * FROM trails WHERE trail_id = :trailId")
    fun getTrail(trailId: String): Flow<TrailEntity?>
    
    @Query("DELETE FROM trails")
    suspend fun clearTable()
}


